src/Nox.Types/Types/Url/Url.cs

```diff
@@ -5,39 +5,48 @@ namespace Nox.Types;
 
 public sealed class Url : ValueObject<System.Uri, Url>
 {
+    private static readonly List<string> ValidSchemes = new()
+    {
+        System.Uri.UriSchemeHttps,
+        System.Uri.UriSchemeHttp,
+        System.Uri.UriSchemeMailto,
+        System.Uri.UriSchemeFtp,
+        System.Uri.UriSchemeFile
+    };
 
     public static Url From(string value)
     {
-        if (!System.Uri.TryCreate(value, System.UriKind.Absolute, out var uriValue))
+        if (!System.Uri.TryCreate(value, UriKind.Absolute, out var uriValue))
         {
             throw new TypeValidationException(
-                new List<ValidationFailure> {
-                    new ValidationFailure("Uri", $"The string '{value}' you provided, is not a valid Uri.") });
+                new List<ValidationFailure> { new(nameof(value), $"The string '{value}' you provided, is not a valid Uri.") }
+            );
         }
 
-        var newObject = new Url
+        var newObject = new Url { Value = uriValue };
+
+        var validationResult = newObject.Validate();
+
+        if (!validationResult.IsValid)
         {
-            Value = uriValue
-        };
+            throw new TypeValidationException(validationResult.Errors);
+        }
 
         return newObject;
     }
 
+    /// <inheritdoc />
     internal override ValidationResult Validate()
     {
         var result = base.Validate();
 
-        bool isValid = System.Uri.TryCreate(base.Value.ToString(), UriKind.Absolute, out var uriResult) &&
-                (uriResult.Scheme == System.Uri.UriSchemeHttps
-                    || uriResult.Scheme == System.Uri.UriSchemeHttp
-                    || uriResult.Scheme == System.Uri.UriSchemeMailto
-                    || uriResult.Scheme == System.Uri.UriSchemeFtp
-                    || uriResult.Scheme == System.Uri.UriSchemeFile);
+        bool isValid = System.Uri.TryCreate(Value.ToString(), UriKind.Absolute, out var uriResult) && ValidSchemes.Contains(uriResult.Scheme);
```

Comment:
Can we remove this part System.Uri.TryCreate(Value.ToString(), UriKind.Absolute, out var uriResult), since it's redundant?